x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -4,101 +4,79 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import fetch from 'node-fetch';
 import yaml from 'js-yaml';
 import { SavedObjectsClientContract } from 'src/core/server/';
-import { Asset, Datasource, Stream } from '../../../ingest/server/libs/types';
+import { Datasource, Stream } from '../../../ingest/server/libs/types';
 import { SAVED_OBJECT_TYPE_DATASOURCES } from '../../common/constants';
 import { AssetReference, Dataset, InstallationStatus, RegistryPackage } from '../../common/types';
+import * as Ingest from '../ingest';
+import { createInput } from '../lib/agent/agent';
 import { CallESAsCurrentUser } from '../lib/cluster_access';
 import { installILMPolicy, policyExists } from '../lib/elasticsearch/ilm/install';
 import { installPipelinesForDataset } from '../lib/elasticsearch/ingest_pipeline/ingest_pipelines';
 import { installTemplateForDataset } from '../lib/elasticsearch/template/install';
 import { getPackageInfo, PackageNotInstalledError } from '../packages';
 import * as Registry from '../registry';
 import { Request } from '../types';
-import { createInput } from '../lib/agent/agent';
 
-export async function createDatasource(options: {
+interface CreateDatasource {
   savedObjectsClient: SavedObjectsClientContract;
   callCluster: CallESAsCurrentUser;
   request: Request;
   pkgkey: string;
   datasourceName: string;
   datasets: Dataset[];
-}) {
-  const { savedObjectsClient, callCluster, pkgkey, datasets, datasourceName, request } = options;
-
+  policyIds: string[];
+}
+export async function createDatasource({
```

Comment:
Tried to clean up `createDatasource` so it's an overview of what happens "create this, install that, etc" and not bothered with the how.

Even with just the top-level instructions it's a pretty long function, but I think this make what's happening more clear. At least it does for me after some time away.

Reply:
+1 on cleaning this up. We need to do more in the future.